### DISCUSSION TITLE
add_middleware is missing from the documentation

### DISCUSSION THREAD

**Discussion by sandeep-gh** [encode/starlette#1762] [Potential Issue] [answered]
Not sure if this is intentional, but add_middleware member function of Starlette() is used quite a few places but is missing from the documentation. Especially on the middleware page (https://www.starlette.io/middleware/). Probably a omission and needs fixing.

**Comment by adriangb** [comment, marked as answer]
This is intentional: https://github.com/encode/starlette/blob/182ed74e28dedb520e40f232e9475efd70b7dde0/starlette/applications.py#L126

Where are you seeing it mentioned?

**Reply by sandeep-gh** [reply]
I see. Not sure exactly where I picked it up from. But the authentication page https://www.starlette.io/authentication/ very last line mentions add_middleware 
`app.add_middleware(AuthenticationMiddleware, backend=BasicAuthBackend(), on_error=on_auth_error)`

**Reply by adriangb** [reply]
Good catch! We removed that line now

**Reply by Kludex** [reply]
For reference, the documentation will have those changes when Starlette 0.21.0 is released.

**Reply by WolfgangFahl** [reply]
Downstream at justpy we'll have to do the changes and I don't understand what is expected:
The comment:
#The following usages are now discouraged in favour of configuration 
seems to expect that some code that is in use today such as:
```python
app.add_middleware(GZipMiddleware)
if SSL_KEYFILE and SSL_CERTFILE:
    app.add_middleware(HTTPSRedirectMiddleware)
```

should be modified. But how - what is the new expected way of achieving the same functionality?

**Reply by adriangb** [reply]
Something like:

```python
middleware = [Middleware(GZipMiddleware)]
if SSL_KEYFILE and SSL_CERTFILE:
    middleware.append(Middleware(HTTPSRedirectMiddleware)
app = Starlette(..., middleware=middleware)
```
